src/dmd/dtoh.d

```diff
@@ -0,0 +1,1256 @@
+                            typeToBuffer(vd.type, vd.ident);
+                        }
+                    }
+                    buf.printf(") {");
+                    foreach (m; *sd.members)
+                    {
+                        if (auto vd = m.isVarDeclaration())
+                        {
+                            if (!memberField(vd))
+                                continue;
+                            buf.printf(" this->%s = %s;", vd.ident.toChars(), vd.ident.toChars());
+                        }
+                    }
+                    buf.printf(" }\n");
+                }
+
+                buf.writestring("};\n\n");
+
+                if (sd.alignment == 1)
+                    buf.writestring("#pragma pack(pop)\n");
+
+                auto savex = buf;
+                buf = checkbuf;
+                buf.writestring("    assert(sizeof(");
+                buf.writestring(sd.ident.toChars());
+                buf.printf(") == %d);\n", sd.size(Loc.initial));
+                buf = savex;
+            }
+            else
+                buf.writestring(";\n\n");
+        }
+
+        private void includeSymbol(AST.Dsymbol ds)
+        {
+            // printf("Forward declaring %s %d\n", ds.toChars(), level);
+            if (cast(void*)ds !in visited)
+            {
+                OutBuffer decl;
+                auto save = buf;
+                buf = &decl;
+                ds.accept(this);
+                buf = save;
+                donebuf.writestring(decl.peekString());
+            }
+        }
+
+        override void visit(AST.ClassDeclaration cd)
+        {
+            if (cast(void*)cd in visited)
+                return;
+            if (cd.getModule() && !cd.getModule().isFrontendModule())
+                return;
+            if (cd.isVisitorClass())
+                return;
+            visited[cast(void*)cd] = true;
+            if (!cd.isCPPclass())
+            {
+                buf.printf("// ignoring non-cpp class %s\n", cd.toChars());
+                return;
+            }
+
+            buf.writestring("class ");
+            buf.writestring(cd.ident.toChars());
+            if (cd.baseClass)
+            {
+                buf.writestring(" : public ");
+                buf.writestring(cd.baseClass.ident.toChars());
+
+                includeSymbol(cd.baseClass);
+            }
+            if (cd.members)
+            {
+                buf.writestring("\n{\npublic:\n");
+                auto save = adparent;
+                adparent = cd;
+                foreach (m; *cd.members)
+                {
+                    m.accept(this);
+                }
+                adparent = save;
+                // Generate special static inline function.
+                if (cd.isIdentifierClass())
+                {
+                    buf.writestring("    static inline Identifier *idPool(const char *s) { return idPool(s, strlen(s)); }\n");
```

Comment:
Why this special casing?

Reply:
Because this doesn't exist in D for obvious reasons.

Reply:
Oh I was more asking: shouldn't we do this in a more general-purpose way.